File: backend/src/main/java/org/globe42/dao/IncomeSourceDao.java
package org.globe42.dao;

import org.globe42.domain.IncomeSource;
import org.springframework.data.jpa.repository.JpaRepository;

/**
 * DAO for the {@link IncomeSource} entity
 * @author JB Nizet
 */
public interface IncomeSourceDao extends JpaRepository<IncomeSource, Long> {
}

File: backend/src/main/kotlin/org/globe42/dao/IncomeSourceDao.kt
package org.globe42.dao

import org.globe42.domain.IncomeSource
import org.springframework.data.jpa.repository.JpaRepository

/**
 * DAO for the [IncomeSource] entity
 * @author JB Nizet
 */
interface IncomeSourceDao : JpaRepository<IncomeSource, Long> {
    fun existsByName(name: String): Boolean
    fun findByName(name: String): IncomeSource?
}
